split-the-stream/README.md

```diff
@@ -0,0 +1,243 @@
+4. How are CI results going to be displayed in dist-git MRs? We wan't to know this so that we can think about ways to take those results and display them for contributors on the source-git MRs.
+
+## Split
+
+We have multiple options:
+
+### 0. no split
+
+- No extra cost of two deployments and two codebases.
+- New jobs will be implemented as new handlers.
+- We don't have support for multiple identities for one forge.
+  - Not hard to do but requires some work.
+- Fedora-source-git friendly: easy combination of events (different for Fedora and Stream)
+  and handlers (=implementation, can be shared).
+
+### 1. same codebase, new deployment
+
+- No extra cost of maintenance of two codebases.
+- New jobs are implemented as new handlers.
+- Different identities can be used in one git forge (=gitlab.com).
+- Resources can be tweaked separately.
+- Fedora-source-git friendly: easy combination of events (different for Fedora and Stream)
+  and handlers (=implementation, can be shared).
+
+### 2. separate workers
+
+- New jobs are implemented as new handlers in a separate repository.
+- The centos-stream related code is in one place, based-on the packit-service code.
+- We can use same or a separate deployment.
+  (Fedora-source-git can be separated or with Stream.)
+
+### 3. split the packit-service repo and build upstream/centos-stream workers
+
+- One repo with the scheduler.
+  Two repositories with the worker(=handlers) definition: one for the upstream, one for the stream.
+- Requires more work.
+- Can lead to a cleaner architecture. Something we were discussing for some time.
+  - The centos-stream related code is in one place,
+    upstream code is in one place and
+    the shared code is in one place.
+- Another dependency in the chain.
+  - It's sometimes hard to work on the functionality that goes across multiple git projects.
+- We can use same or a separate deployment.
+  (Fedora-source-git can be separated or with Stream.)
+
+### 4. fork and improve
+
+- The benefits of the current service code can be preserved.
+- The non-relevant/bad code can be removed.
+- More time needed for development.
+- Improvements relevant for both are hard to sync.
+- More time needed for maintenance.
+
+### 5. separate project from scratch
+
+- The new service can be more lightweight and efficient.
+- We can iterate on the prototype more quickly.
+- We can get rid of the old bad staff in the packit-service.
+- We can go through the same pain we've already gone through.
+- We need to maintain two separate projects. (We need more people or reduce the productivity.)
+- We are not motivated to improve the current projects.
+- To share the code between the upstream and source project, we need to create some shared libraries.
+  - Can lead to 3. and/or having another project on our dependency chain.
+
+### Database
```

Comment:
I'm wondering what would be the common parts to be shared between the Packit and source-git use-cases? And how much would they represent from the total schema definition for the Packit and source-git use-case?

In the source-git usecase we would need to store data for the following types of pipelines:
- pipelines triggered by source-git MRs, which will create/update dist-git MRs;
- pipelines triggered by dist-git MRs being merged, which will merge the corresponding source-git MRs;
- pipelines triggered by other dist-git changes, which will open source-git MRs.

Of course the functionality above might change, based on what solutions we choose, but nevertheless the above could be used to tell what parts of the current schema could be used to support the above use cases. At first sight, it seems to me, there is no commonality, but I might be terribly wrong about this :smiley:

Reply:
Thanks for the comment. I've covered this in the text and put it on the agenda also for the next arch. meeting so we can agree on something.